src/argilla/utils/modeling.py

```diff
@@ -0,0 +1,225 @@
+#  coding=utf-8
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import re
+from typing import Dict, List, Optional, Tuple, Union
+
+import pandas as pd
+import textdescriptives as td
+from tqdm import tqdm
+
+from argilla.client.feedback.dataset.local.dataset import FeedbackDataset
+from argilla.client.feedback.schemas.metadata import (
+    FloatMetadataProperty,
+    IntegerMetadataProperty,
+    TermsMetadataProperty,
+)
+from argilla.client.feedback.schemas.records import FeedbackRecord
+
+
+class TextDescriptivesExtractor:
+    """This class extracts a number of basic text descriptives from FeedbackDataset
+    records using the TextDescriptives library and adds them as record metadata."""
+
+    def __init__(
+        self,
+        model: str = "en",
+        metrics: list[str] = ["quality", "readability", "information_theory"],
```

Comment:
It is good to take the mutability of lists into account.
https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

Reply:
@davidberenstein1957 If we replace the default value for `metrics` with a sentinel value (`metrics: Optional[List[str]] = None`) as suggested in the link above, then the user could instantiate the class without providing any metrics. Don't we want to force the user to provide a set of metrics? In which case, should I change the default value for `metrics` to immutable tuples (`metrics: Tuple[str, str, str] = ("quality", "readability", "information_theory")`)?

Reply:
I think we first need to set it as None and later on define the metrics `if metrics == None` then define it.

Reply:
@davidberenstein1957 I agree, I couldn't find a clear mapping of metrics and their corresponding submetrics. If we want to allow the user to specify the submetrics, then I think we will need a mapping of metrics and their corresponding submetrics. For the mapping we could:
1. Define the mapping manually, which would be the most straightforward but potentially break if significant changes are made the to text-descriptives library.
2. Programmatically create a mapping by extracting all the metrics and submetrics from a sample string like this:
```python
nlp = spacy.load("en_core_web_sm")
nlp.add_pipe("textdescriptives/all")
doc = nlp("This is a sentence.")
dict_attrs = [attr for attr in dir(doc._) if isinstance(getattr(doc._, attr), dict)]
dict_keys = {attr: list(getattr(doc._, attr).keys()) for attr in dict_attrs}
```
Which gives us a mapping like this:
```
{'_quality': ['n_stop_words',
  'alpha_ratio',
  'mean_word_length',
  'doc_length',
  'symbol_to_word_ratio',
  'proportion_ellipsis',
  'proportion_bullet_points',
  'contains',
  'duplicate_line_chr_fraction',
  'duplicate_paragraph_chr_fraction',
  'duplicate_ngram_chr_fraction',
  'top_ngram_chr_fraction',
  'oov_ratio'],
 'coherence': ['first_order_coherence', 'second_order_coherence'],
 'counts': ['n_tokens',
  'n_unique_tokens',
  'proportion_unique_tokens',
  'n_characters',
  'n_sentences'],
...}
```

Reply:
I would go for the sample string one.

Reply:
After trying this approach, I found that there were inconsistencies and too much complexity in trying to programmatically fetch a map of text-descriptives metrics and submetrics. Given the vast number of submetrics to choose from and the amount of complexity it would add to the codebase, I recommend specifying only metrics (as opposed to submetrics) for the initial release of this feature.